### PR TITLE
[cli] add CLI commands for testing `otPlatTcp` APIs

### PR DESCRIPTION
**Pull request by abtink**
This commit introduces a new `plattcp` CLI command designed primarily for testing and evaluating `otPlatTcp` platform implementations. 

The new CLI provides commands to manually drive the platform TCP API.

These commands act as a thin wrapper around the `Ip6::PlatTcp` classes, allowing to manually invoke and observe connection events, states, and data reception from a console. The feature is guarded by `OPENTHREAD_CONFIG_CLI_PLAT_TCP_ENABLE`.

---------

Here is example of test output (on two posix devices with simulated RCP):

Tab1:

```bash
$ sudo ./src/posix/ot-cli 'spinel+hdlc+forkpty://examples/apps/ncp/ot-rcp?forkpty-arg=1'

> plattcp listener enable fe80::10aa:a3b9:93f2:4462 1234 14
Enabling listener on [fe80:0:0:0:10aa:a3b9:93f2:4462%14]:1234
Done
 
> plattcp listener
Enabled
Done
```

Tab2:

```bash
$ sudo ./src/posix/ot-cli 'spinel+hdlc+forkpty://examples/apps/ncp/ot-rcp?forkpty-arg=2'

> plattcp connect fe80::10aa:a3b9:93f2:4462 1234 14
Connecting to [fe80:0:0:0:10aa:a3b9:93f2:4462%14]:1234
Done
PlatTcp:: HandleEvent Connected
```

And we see this on Tab1

```bash
PlatTcp: Accept [fe80:0:0:0:10aa:a3b9:93f2:4462%14]:56071
PlatTcp:: HandleEvent Connected
```

Send and recv:

On tab2:
```
> plattcp send 32
Done
PlatTcp:: HandleEvent SendDone
```

On tab 1:
```
PlatTcp:: HandleEvent Receive
PlatTcp: Received 32 bytes
    000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f
```    

On tab2:

```
> plattcp close
PlatTcp:: HandleEvent Disconnected
PlatTcp:: DisconnectReason: Closed
Done
```

And on tab1:
```
PlatTcp:: HandleEvent Disconnected
PlatTcp:: DisconnectReason: Closed
```

### CODE REVIEW

File: src/cli/cli_plat_tcp.cpp [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    uint32_t ifIndex;
+
+    aSockAddr.Clear();
+
+    VerifyOrExit(!aArgs[0].IsEmpty());
+    SuccessOrExit(error = aArgs[0].ParseAsIp6Address(aSockAddr.GetAddress()));
+
+    VerifyOrExit(!aArgs[1].IsEmpty());
+    SuccessOrExit(error = aArgs[1].ParseAsUint16(port));
+    aSockAddr.SetPort(port);
```
[the remark this answers is not in the document]
Comment (abtink) [reply src/cli/cli_plat_tcp.cpp:141]: That is intentional. If they are omitted, we keep the zero value!

[the remark this answers is not in the document]
Comment (jwhui) [reply src/cli/cli_plat_tcp.cpp:141]: Will this be confusing with something like `plattcp connect`? It will try to use zero address/port for the peer.

[the remark this answers is not in the document]
Comment (abtink) [reply src/cli/cli_plat_tcp.cpp:141]: We can add additional code to check for specific cases, but I'm not sure it adds much value here. 

The `plattcp` CLI is only intended for testing the platform implementation, so we probably don't need to be overly strict or specific about command validation in CLI itself. E.g., it could be useful to test certain corner cases (e.g., connecting with an invalid or zero port).

File: src/cli/cli_plat_tcp.cpp [hunk 2]
```diff
@@ -0,0 +1,302 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
```
Comment (jwhui) [inline src/cli/cli_plat_tcp.cpp]:
```suggestion
 *  Copyright (c) 2026, The OpenThread Authors.
```

**Review by jwhui — APPROVED** [review]
